### ISSUE TITLE
ERROR: Aborted with Exit Code 1

### ISSUE THREAD

**Issue by MiDWalkeR** [ThrowTheSwitch/Ceedling#170]
Hi!

Seems like I need your suggestion. 
What I have done:

1. ceedling new ceed_my

2. goto **src** folder and create two files:
foo.c
```
#include <stdio.h>

int foo(int val)
{
    return val;
}
```
foo.h
`int foo(int val);`

3. goto **test** folder and create file:
test_foo.c

```
#include <stdio.h>
#include "foo.h"
#include "unity.h"

void setUp(void)
{
}

void tearDown(void)
{

}

void test_foo_function(void)
{
    TEST_ASSERT_MESSAGE(foo(0), "oops");    
}
```
In case of foo(1) test report looks good, like:
```
--------------------
OVERALL TEST SUMMARY
--------------------
TESTED:  1
PASSED:  1
FAILED:  0
IGNORED: 0
```

But if foo(0), I got 
```
---------------------
BUILD FAILURE SUMMARY
---------------------
Unit test failures.

ERROR: Aborted with Exit Code 1
```
I mean, here is no information about what and where was broken, like 
`test_foo.c:16:test_foo_function:FAIL:oops`

Could you please give me some advice, what I'm doing wrong? Thanks.

**Comment by Ouss4** [comment]
TEST_ASSERT() tests the value of the boolean passed in its arguments.
If its true (=1), the test passes, if its false(=0), you will have an error, exactly like the one you have. (see http://www.throwtheswitch.org/unity/ scroll down a little bit.)

If you are trying to test for integers use `TEST_ASSERT_EQUAL_INT(ExptectedValue, YourValue);`
That would give something like this in your case:
```

TEST_ASSERT_EQUAL_INT(1, foo(1)); // Test passes
TEST_ASSERT_EQUAL_INT(0, foo(0)); // Test passes
TEST_ASSERT_EQUAL_INT(1, foo(0)); // Test fails.
```

**Comment by MiDWalkeR** [comment]
Nope, the same behaviour:

I used another MACRO for int values, as you said: TEST_ASSERT_EQUAL_INT.  If Test passed I get good overal test summary, otherwise:
```
---------------------
BUILD FAILURE SUMMARY
---------------------
Unit test failures.

ERROR: Aborted with Exit Code 1
```

**Comment by Letme** [comment]
That seems like patch for returning error code prevents Overall summary printout? @dpostorivo I think you might help here...

**Comment by MiDWalkeR** [comment]
I forgot to add: All this manupilation I made under Windows 10, using msys2 + tmux.

**Comment by mvandervoord** [comment]
If you type `gem list`, what does the ceedling line tell you?

If you open your project.yml file, what does the :plugins section say? 

I ask because the message you are seeing is reported only when a plugin has failed to run. Knowing the version of ceedling and the settings could help us debug it. I went through the procedure you listed and can't reproduce it at the moment, but my platform is admittedly different than yours at the moment. I'll try to test it on a Windows 10 VM when I get a chance.

**Comment by MiDWalkeR** [comment]
Hi, @mvandervoord

> If you type gem list, what does the ceedling line tell you?

```
bigdecimal (1.2.8)
ceedling (0.27.0)
did_you_mean (1.0.0)
io-console (0.4.5)
json (1.8.3)
minitest (5.8.5)
net-telnet (0.1.1)
power_assert (0.2.6)
psych (2.1.0)
rake (10.4.2)
rdoc (4.2.1)
test-unit (3.1.5)
thor (0.19.4)
```

> If you open your project.yml file, what does the :plugins section say?

```
:plugins:
  :load_paths:
    - vendor/ceedling/plugins
  :enabled:
    - stdout_pretty_tests_report
    - module_generator
```

Also, if I'm using unity by itself by including unity.h and make compilation by my own, everything looks good.

Thanks for your support, guys.

**Comment by dpostorivo** [comment]
@MiDWalkeR out of curiosity, what command exactly are you running to run the tests?

- rake test
- rake test:all
- ceedling test
- ceedling test:all
- something else?

**Comment by MiDWalkeR** [comment]
Ceedling test:all

**Comment by dpostorivo** [comment]
See if you get the same result using `rake test:all`. I don't think that will fix it, but I've noticed different behavior ceedling vs rake as seen in #148 so there's a slim chance it's similar.

**Comment by MiDWalkeR** [comment]
Hi, guys!

>  I don't think that will fix it
But It was:)

With a `rake test:all` everything works fine. Seems like I should use only that command instead of ceedling. 

Thank you, @dpostorivo

**Comment by mvandervoord** [comment]
Directly calling rake is the old method and is definitely going away soon. Clearly we're not quite there for that, yet.

Just verifying: This was a clean install, correct? It appears from your first message that you directly installed the 0.27.0 gem and then created the project with that.

**Comment by dpostorivo** [comment]
@mvandervoord  I understand why you would want to move away from `rake` just to using `ceedling`, but until both produce the same exact behaviour from execution to runtime to exit status it's not feasible. 
This issue and #148 show drastically different behaviours that must be corrected and these issues have been pretty.

As a side note, a lot of new errors have been reported in 0.27.0 including #147,  #159, #163, and #166 and 0.25.0 (the previous version) had the issue where `rake test:all` would return 0 even if tests fail so I'm stuck a few versions back and would really like to upgrade (which is why I've been investing in the module testing recently). I suggest once those issues are resolved doing a patch release and then if we want to focus on getting the ceedling executable in a better place we can and carve out what the goal of the next few releases should be so other willing contributors can help get ceedling there.

**Comment by MiDWalkeR** [comment]
@mvandervoord ceedling was installed following the description here: http://www.throwtheswitch.org/ceedling

**Comment by TAGC** [comment]
I get the same problem:

```YAML
$ ceedling
---------------------
BUILD FAILURE SUMMARY
---------------------
Unit test failures.

ERROR: Aborted with Exit Code 1
```

Extremely unhelpful. Here's my `project.yml`:

```
:project:
  :use_exceptions: FALSE
  :use_test_preprocessor: FALSE
  :use_auxiliary_dependencies: TRUE
  :build_root: build
  :release_build: FALSE
  :test_file_prefix: Test

:environment:
  - :path:
    - 'C:/Program Files (x86)/IAR Systems/Embedded Workbench 6.5/arm/bin'
    - 'C:/Program Files (x86)/IAR Systems/Embedded Workbench 6.5/common/bin'
    - #{ENV['PATH']}

:extension:
  :executable: .out

:paths:
  :test:
    - +:test/**
    - -:test/support
  :source:
    - src/**
                         
                           
  :support:
    - test/support

:defines:
  :commmon: &common_defines []
  :test:
    - *common_defines
    - TEST
  :test_preprocess:
    - *common_defines
    - TEST

:cmock:
                     
  :when_no_prototypes: :warn
  :enforce_strict_ordering: TRUE
  :plugins:
    - :ignore
    - :callback
  :treat_as:
    uint8:    HEX8
    uint16:   HEX16
    uint32:   UINT32
    int8:     INT8
    bool:     UINT8

:tools:
  :test_compiler:
    :executable: iccarm
    :name: 'IAR test compiler'
    :arguments:
      - -D _DLIB_FILE_DESCRIPTOR=1
      - --debug
      - --endian=little
      - --cpu=Cortex-M3
      - -e
      - --fpu=None
      - -Ol
      - --preprocess "${3}"
      - --dlib_config "C:/Program Files (x86)/IAR Systems/Embedded Workbench 6.5/arm/INC/c/DLib_Config_Normal.h"
      - -I"$": COLLECTION_PATHS_TEST_TOOLCHAIN_INCLUDE
      - -I"$": COLLECTION_PATHS_TEST_SUPPORT_SOURCE_INCLUDE_VENDOR
      - -o "${2}"
      - --diag_suppress=Pa050
      - '"${1}"'
    
  :test_linker:
    :executable: ilinkarm
    :name: 'IAR test linker'
    :arguments:
      - --vfe
      - --redirect _Printf=_PrintfFull
      - --redirect _Scanf=_ScanfFull
      - --semihosting
      - --config "C:/Program Files (x86)/IAR Systems/Embedded Workbench 6.5/arm/config/generic_cortex.icf"
      - --map "${3}"
      - -o "${2}"
      - '"${1}"'
      
  :test_fixture:
    :executable: cspybat
    :name: 'CSpyBat test runner'
    :arguments:
      - '"C:/Program Files (x86)/IAR Systems/Embedded Workbench 6.5/arm/bin/armproc.dll"'
      - '"C:/Program Files (x86)/IAR Systems/Embedded Workbench 6.5/arm/bin/armsim2.dll"'
      - '"${1}"'
      - --plugin "C:/Program Files (x86)/IAR Systems/Embedded Workbench 6.5/arm/bin/armbat.dll"
      - --backend -B
      - --endian=little
      - --cpu=Cortex-M3
      - --fpu=None
      - --semihosting

# LIBRARIES
# These libraries are automatically injected into the build process. Those specified as
# common will be used in all types of builds. Otherwise, libraries can be injected in just
# tests or releases. These options are MERGED with the options in supplemental yaml files.
:libraries:
  :placement: :end
  :flag: "${1}"  # or "-L ${1}" for example
  :common: &common_libraries []
  :test:
    - *common_libraries
  :release:
    - *common_libraries

:plugins:
  :load_paths:
    - vendor/ceedling/plugins
  :enabled:
    - stdout_pretty_tests_report
    - module_generator
```

**Comment by TAGC** [comment]
Updating my `project.yml` to use the default GCC compiler (instead of IAR's compiler), I still see the same problem. However, if I replace the `vendor` folder with the `vendor` folder used by the `temp_sensor` example project, I get the correct output.

To clarify what I mean. First I use the set up the following `project.yml`:

```YAML
:project:
  :use_exceptions: FALSE
  :use_test_preprocessor: TRUE
  :use_auxiliary_dependencies: TRUE
  :build_root: build
  :release_build: FALSE
  :test_file_prefix: Test

:extension:
  :executable: .out

:paths:
  :test:
    - +:test/**
    - -:test/support
  :source:
    - src/**
  :support:
    - test/support

:defines:
  :commmon: &common_defines []
  :test:
    - *common_defines
    - TEST
  :test_preprocess:
    - *common_defines
    - TEST

:cmock:
  :when_no_prototypes: :warn
  :enforce_strict_ordering: TRUE
  :plugins:
    - :ignore
    - :callback
    - :expect_any_args
    - :array
  :treat_as:
    uint8:    HEX8
    uint16:   HEX16
    uint32:   UINT32
    int8:     INT8
    bool:     UINT8

:libraries:
  :placement: :end
  :flag: "${1}"  # or "-L ${1}" for example
  :common: &common_libraries []
  :test:
    - *common_libraries
  :release:
    - *common_libraries

:plugins:
  :load_paths:
    - vendor/ceedling/plugins
  :enabled:
    - stdout_pretty_tests_report
    - module_generator
```

Then I try running `ceedling version` with the latest `vendor` folder under my project directory:

```
  Ceedling:: 0.27.0
CException:: 1.3.1.18
     CMock:: 2.4.4.215
     Unity:: 2.4.1.120
```

Then I try running `ceedling test:all`:

```
---------------------
BUILD FAILURE SUMMARY
---------------------
Unit test failures.

ERROR: Aborted with Exit Code 1
```

I then move `vendor` -> `vendor_bak`, and copy `vendor` over from `temp_sensor` into my project.

I run `ceedling version` again:

```
  Ceedling:: 0.25.0
CException:: 1.3.1.18
     CMock:: 2.4.3.215
     Unity:: 2.4.0.120
```

And `ceedling test:all` again:

```
---------------------
BUILD FAILURE SUMMARY
---------------------
Unit test failures.



Test 'TestCanHandler.c'
-----------------------
Creating mock for CanHardware...
Generating runner for TestCanHandler.c...
Compiling TestCanHandler_runner.c...
Compiling TestCanHandler.c...
Compiling MockCanHardware.c...
Compiling unity.c...
Compiling CanHandler.c...
Compiling cmock.c...
Linking TestCanHandler.out...
Running TestCanHandler.out...

-----------
TEST OUTPUT
-----------
[TestCanHandler.c]
  - ""

-------------------
FAILED TEST SUMMARY
-------------------
[TestCanHandler.c]
  Test: test_CAN_Handler_Publishes_Messages_Received_On_Source_Port
  At line (57): "Expected 256 Was 0"

--------------------
OVERALL TEST SUMMARY
--------------------
TESTED:  2
PASSED:  1
FAILED:  1
IGNORED: 0
```

**Comment by mvandervoord** [comment]
That must be an older version of the temp_sensor project that you installed awhile ago, yes? (noting the 0.25.0 version when you copied it over).

If you feel like experimenting, we have an ongoing discussion on this topic on issue #174

**Comment by TAGC** [comment]
> That must be an older version of the temp_sensor project that you installed awhile ago, yes?

I believe so, yeah.

**Comment by mvandervoord** [comment]
Fixed in release 0.28.1.  Don't forget to do a "ceedling update PROJECTNAME" from the folder outside your project after you install the latest gem.
